### ISSUE TITLE
Region map: k-means choropleth coloring breaks with small number of rows

### ISSUE THREAD

**Issue by tlrobinson** [metabase/metabase#4684]
If the number of rows is less than the number of colors (5), the k-means library throws this error:

> ckmeans.js?7bb6:223 Uncaught Error: Cannot generate more classes than there are data values

**Comment by tanzoniteblack** [comment]
Whoops, I'll have a fix for that shortly.

**Comment by mariomka** [comment]
I have the same problem in v0.31.0.

**Comment by FadhelC** [comment]
Same issue in v0.31.1

**Comment by programwar** [comment]
It's still present on v0.31.2

**Comment by clouless** [comment]
I am getting the exact same error Dockerized Metabase `v0.31.2`.

I have a column for ISO 31661 a2 country codes and a sum of sales.

![image](https://user-images.githubusercontent.com/12599965/54754899-eea61800-4be4-11e9-9e7d-b83223241ab1.png)

When I change the view to map, then this happens:

![demo](https://user-images.githubusercontent.com/12599965/54755116-7b50d600-4be5-11e9-8dc7-a7212a024fe8.gif)

I get the console error:
```
vendor.bundle.js?78262faa1c529a733077:189 Uncaught Error: cannot generate more classes than there are data values
    at Object.t.exports [as ckmeans] (vendor.bundle.js?78262faa1c529a733077:189)
    at t.value (app-main.bundle.js?78262faa1c529a733077:5)
    at u._renderValidatedComponentWithoutOwnerOrContext (vendor.bundle.js?78262faa1c529a733077:145)
    at u._renderValidatedComponent (vendor.bundle.js?78262faa1c529a733077:145)
    at u._updateRenderedComponent (vendor.bundle.js?78262faa1c529a733077:145)
    at u._performComponentUpdate (vendor.bundle.js?78262faa1c529a733077:145)
    at updateComponent (vendor.bundle.js?78262faa1c529a733077:145)
    at u.performUpdateIfNecessary (vendor.bundle.js?78262faa1c529a733077:145)
    at Object.performUpdateIfNecessary (vendor.bundle.js?78262faa1c529a733077:83)
    at b (vendor.bundle.js?78262faa1c529a733077:83)
```

I found no workaround to get this working. Does not matter for me if Cumulative Sum or just Sum.

 * Metabase: dockerized `metabase/metabase:v0.32.0-RC1` and `metabase/metabase:v0.31.2`
 * Browser: Google Chrome v72.0.3626.121 (official build) (64-Bit)
 * OS: macOS Mojave v10.14.1
 * Database: dockerized postgres:10-alpine
 * SQL Schema and testdata

```sql
CREATE USER metabase WITH PASSWORD 'metabase123';

DROP DATABASE IF EXISTS "stats";
CREATE DATABASE "stats"
    WITH OWNER "metabase"
    ENCODING 'UTF8'
    LC_COLLATE = 'en_US.UTF-8'
    LC_CTYPE = 'en_US.UTF-8'
    TEMPLATE template0;

\c stats;

--
-- TABLES
--
DROP TABLE IF EXISTS atlassian_marketplace_products CASCADE;
CREATE TABLE atlassian_marketplace_products(
 id serial PRIMARY KEY,
 name VARCHAR(255) NOT NULL,
 app_key VARCHAR(255) NOT NULL
);
DROP TABLE IF EXISTS atlassian_marketplace_app_types CASCADE;
CREATE TABLE atlassian_marketplace_app_types(
 id serial PRIMARY KEY,
 name VARCHAR(255) NOT NULL
);
DROP TABLE IF EXISTS atlassian_marketplace_sales_per_country CASCADE;
CREATE TABLE atlassian_marketplace_sales_per_country(
 id serial PRIMARY KEY,
 product_fk INTEGER REFERENCES atlassian_marketplace_products(id),
 num_sales INTEGER NOT NULL,
 country VARCHAR(255) NOT NULL,
 country_iso VARCHAR(255) NOT NULL,
 revenue  DECIMAL NOT NULL,
 app_type_fk INTEGER REFERENCES atlassian_marketplace_app_types(id),
 timestamp TIMESTAMPTZ NOT NULL
);

--
-- DEMO DATA
--
INSERT INTO atlassian_marketplace_products(name, app_key) VALUES('App1', 'app1key');

INSERT INTO atlassian_marketplace_app_types(name) VALUES('Cloud');
INSERT INTO atlassian_marketplace_app_types(name) VALUES('Server');
INSERT INTO atlassian_marketplace_app_types(name) VALUES('Data Center');
INSERT INTO atlassian_marketplace_app_types(name) VALUES('All');

INSERT INTO atlassian_marketplace_sales_per_country(product_fk, num_sales, country, country_iso, revenue, app_type_fk, timestamp)
    VALUES(1, 2, 'Germany', 'DE', 500, 1, '2019-03-17 12:00:00.000000 EST');

INSERT INTO atlassian_marketplace_sales_per_country(product_fk, num_sales, country, country_iso, revenue, app_type_fk, timestamp)
    VALUES(1, 3, 'United States of America', 'US', 500, 1, '2019-03-17 12:00:00.000000 EST');

--
-- PERMISSIONS
--
GRANT ALL PRIVILEGES ON TABLE atlassian_marketplace_products TO metabase;
GRANT ALL PRIVILEGES ON TABLE atlassian_marketplace_app_types TO metabase;
GRANT ALL PRIVILEGES ON TABLE atlassian_marketplace_sales_per_country TO metabase;

GRANT ALL PRIVILEGES ON SEQUENCE atlassian_marketplace_products_id_seq TO metabase;
GRANT ALL PRIVILEGES ON SEQUENCE atlassian_marketplace_app_types_id_seq TO metabase;
GRANT ALL PRIVILEGES ON SEQUENCE atlassian_marketplace_sales_per_country_id_seq TO metabase;
```

And more testdata (does not fix it)

```sql
INSERT INTO atlassian_marketplace_sales_per_country(product_fk, num_sales, country, country_iso, revenue, app_type_fk, timestamp)
    VALUES(1, 2, 'Jamaica', 'JM', 999, 1, '2019-02-17 12:00:00.000000 EST');

INSERT INTO atlassian_marketplace_sales_per_country(product_fk, num_sales, country, country_iso, revenue, app_type_fk, timestamp)
    VALUES(1, 3, 'Netherlands', 'NL', 999, 1, '2019-02-17 12:00:00.000000 EST');

INSERT INTO atlassian_marketplace_sales_per_country(product_fk, num_sales, country, country_iso, revenue, app_type_fk, timestamp)
    VALUES(1, 2, 'Jamaica', 'JM', 999, 1, '2019-01-17 12:00:00.000000 EST');

INSERT INTO atlassian_marketplace_sales_per_country(product_fk, num_sales, country, country_iso, revenue, app_type_fk, timestamp)
    VALUES(1, 3, 'Netherlands', 'NL', 999, 1, '2019-01-17 12:00:00.000000 EST');

INSERT INTO atlassian_marketplace_sales_per_country(product_fk, num_sales, country, country_iso, revenue, app_type_fk, timestamp)
    VALUES(1, 2, 'Jamaica', 'JM', 999, 1, '2018-12-17 12:00:00.000000 EST');

INSERT INTO atlassian_marketplace_sales_per_country(product_fk, num_sales, country, country_iso, revenue, app_type_fk, timestamp)
    VALUES(1, 3, 'Netherlands', 'NL', 999, 1, '2018-12-17 12:00:00.000000 EST');


INSERT INTO atlassian_marketplace_sales_per_country(product_fk, num_sales, country, country_iso, revenue, app_type_fk, timestamp)
    VALUES(1, 2, 'Jamaica', 'JM', 999, 1, '2019-02-17 12:00:00.000000 EST');

INSERT INTO atlassian_marketplace_sales_per_country(product_fk, num_sales, country, country_iso, revenue, app_type_fk, timestamp)
    VALUES(1, 3, 'Netherlands', 'NL', 999, 1, '2019-02-17 12:00:00.000000 EST');

INSERT INTO atlassian_marketplace_sales_per_country(product_fk, num_sales, country, country_iso, revenue, app_type_fk, timestamp)
    VALUES(1, 2, 'Jamaica', 'JM', 999, 1, '2019-01-17 12:00:00.000000 EST');

INSERT INTO atlassian_marketplace_sales_per_country(product_fk, num_sales, country, country_iso, revenue, app_type_fk, timestamp)
    VALUES(1, 3, 'Netherlands', 'NL', 999, 1, '2019-01-17 12:00:00.000000 EST');

INSERT INTO atlassian_marketplace_sales_per_country(product_fk, num_sales, country, country_iso, revenue, app_type_fk, timestamp)
    VALUES(1, 2, 'Jamaica', 'JM', 999, 1, '2018-12-17 12:00:00.000000 EST');

INSERT INTO atlassian_marketplace_sales_per_country(product_fk, num_sales, country, country_iso, revenue, app_type_fk, timestamp)
    VALUES(1, 3, 'Netherlands', 'NL', 999, 1, '2018-12-17 12:00:00.000000 EST');
```

**Comment by fvides** [comment]
Para nada

      	
Francisco Vides Fernández / Gerente
fvides@idedalo.com
DÉDALO / INGENIEROS EN INFORMÁTICA
Tel: 952 602 959
Calle Afligidos, 2, 29015 Málaga
www.idedalo.com



En 21 de marzo de 2019 15:15:55 Bernhard Grünewaldt

**Comment by RobertLowe** [comment]
I can report this happening as well in 0.31.2 with full country names "Canada", "United States" etc

**Comment by luisberns** [comment]
I'm having this issue in v0.32.8

With the Custom and Native query. The map is shown normally when there's no filter, but when I filter and the results are less than 5 rows the map just keep loading.

---
Query: Region Map in the Brazil area

Browser: Chrome v74.0.3729.169
OS: macOS Mojave v10.14.5
Database: MySQL
Metabase: v0.32.8

**Comment by flamber** [comment]
@luisberns This is not fixed in 0.32.8 - it will be fixed in next release, probably 0.33
